engines/mohawk/myst_stacks/mechanical.cpp

```diff
@@ -846,8 +841,6 @@ void Mechanical::o_fortressRotation_init(uint16 var, const ArgumentsArray &args)
 	uint32 movieDuration = gears->getDuration().convertToFramerate(600).totalNumberOfFrames();
 	if (movieDuration == 3680) {
 		_fortressRotationShortMovieWorkaround = true;
-		_fortressRotationShortMovieCount = 0;
```

Comment:
Is removing those two lines necessary? It would seem preferable to me if the puzzle was fully reset when accessing it.

Reply:
movieDuration == 3680 is only true when the fortress was last set to West. So what the setting of zero for these two variables does it causes the direction that the rotator is set to to be reset to the default/starting direction (South), but again this only happens when the direction was last West. 

So the behavior before this PR is that the Rotator direction will be the last location it was unless it was West then it will be South when the user steps back up to the rotator.

We can have the puzzle reset everytime the user returns to the puzzle, but that is not what these two lines of code (with the if check) does and it is a deviation from what regular Myst does. The West direction is currently behaving differently then all the other directions.

The only deviation that this PR makes is that if the user changes the direction, but does not lower the left stick (to lock on) and hits the button the direction the fortress is pointing will not be this new direction, were as the original would have it be this new direction.

Reply:
> movieDuration == 3680 is only true when the fortress was last set to West.

Are you sure about that ? That test is meant to detect the broken video file that shipped with Myst ME. It should be true regardless of the position.

Reply:
Okay, sorry let me clarify.  

Mechanical::o_fortressRotation_init is called whenever the player steps back up to the rotator puzzle controls. For Myst ME the condition is true so _fortressRotationShortMovieCount = 0; this variable being set to zero causes the moviePosition value in Mechanical::fortressRotation_run() to map to South since the gears are not looping (since the player just stepped up to the controls). 

In this case the moviePosition should of been retained since the gears were not just turning, but since _fortressRotationShortMovieCount is set to zero this direction state is lost. I am not sure why this is not also an issue for north.

If you set the direction to west. step away from the controls and back to it the controls will be still set to west (can be checked by raising and lowering the left control stick) with this PR, but it will reset to south before this PR.

Reply:
Ok. The catch with `_fortressRotationShortMovieCount` and `_fortressRotationShortMovieLast` is that they are not persisted in the saved state. So, if you save and restore, the same problem as before your fix should happen. Perhaps those variables need to be initialized in `o_fortressRotation_init` to values that will make `fortressRotation_run` work.

Reply:
The player always starts on the south island when loading a save in mech age so the direction always being set to south on init (and not stored in save state) makes sense.